src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -46,6 +63,30 @@ public void storeMessage(Message message) {
     datastore.put(messageEntity);
   }
 
+  /** Stores the User in Datastore */
+  public void storeUser(String user) {
+    Entity userEntity = new Entity("User", user);
+    // Faster to compare ints than strings, use hash to check if user in Datastore
+    userEntity.setProperty("hashcode", user.hashCode());
+    if (userFound(user.hashCode()) != 0) {
```

Comment:
I also don't think this is accurate, because two different objects can have the same hash code.  The only real requirement is that the same objects have the same hash code, but there's no guarantee about the distribution of hash values.  I think you should just use the user id (what you call String user here).